projects/step-frontend/src/lib/modules/timeseries/performance-view/filter-bar/dashboard-filter-bar.component.ts

```diff
@@ -278,7 +278,7 @@ export class DashboardFilterBarComponent implements OnInit, OnDestroy {
     const itemToDelete = this._internalFilters[index];
 
     this._internalFilters.splice(index, 1);
-
+    this._internalFilters = [...this._internalFilters];
```

Comment:
1. Remove underscore for `internalFilters` as it's not an injection
2. Remove and reference update can be optimized in one action:
```
this.internalFilters = this.internalFilters.filter((item) => item !== itemToDelete);
```
It will remove the item and create a new array reference.